convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,205 @@
+
+class EfibootmgrUtilityInstalled(actions.Action):
+    id = "EFIBOOTMGR_UTILITY_INSTALLED"
+    dependencies = ("NEW_DEFAULT_EFI_BIN",)
+
+    def run(self):
+        """Check if the Efibootmgr utility is installed"""
+        super(EfibootmgrUtilityInstalled, self).run()
+
+        if not os.path.exists("/usr/sbin/efibootmgr"):
+            self.set_result(
+                level="ERROR",
+                id="EFIBOOTMGR_UTILITY_NOT_INSTALLED",
+                title="Efibootmgr utility is not installed",
+                description="The /usr/sbin/efibootmgr utility is not installed.",
+                remediations="Install the efibootmgr utility via YUM/DNF.",
+            )
+
+
+class CopyGrubFiles(actions.Action):
+    id = "COPY_GRUB_FILES"
+    dependencies = ("EFIBOOTMGR_UTILITY_INSTALLED",)
+
+    def run(self):
+        """Copy grub files from centos/ dir to the /boot/efi/EFI/redhat/ dir.
+
+        The grub.cfg, grubenv, ... files are not present in the redhat/ directory
+        after the conversion on a CentOS Linux system. These files are usually created
+        during the OS installation by anaconda and have to be present in the
+        redhat/ directory after the conversion.
+
+        The copy of the centos/ directory should be ok. In case of the conversion
+        from Oracle Linux, the redhat/ directory is already used.
+        """
+        super(CopyGrubFiles, self).run()
+
+        if systeminfo.system_info.id != "centos":
+            logger.debug("Skipping copying GRUB files - only related to CentOS Linux.")
+            return
+
+        # TODO(pstodulk): check behaviour for efibin from a different dir or with a different name for the possibility of
+        #  the different grub content...
+        # E.g. if the efibin is located in a different directory, are these two files valid?
+        logger.info("Copying GRUB2 configuration files to the new UEFI directory %s." % RHEL_EFIDIR_CANONICAL_PATH)
+        required = ["grubenv", "grub.cfg"]
+        optional = ["user.cfg"]
+        all_files = required + optional
+        for filename in all_files:
+            src_path = os.path.join(CENTOS_EFIDIR_CANONICAL_PATH, filename)
+            dst_path = os.path.join(RHEL_EFIDIR_CANONICAL_PATH, filename)
+            if os.path.exists(dst_path):
+                logger.debug("The %s file already exists. Copying skipped." % dst_path)
+                continue
+            if not os.path.exists(src_path):
+                if filename in required:
+                    # without the required files user should not reboot the system
+                    self.set_result(
+                        level="ERROR",
+                        id="UNABLE_TO_FIND_REQUIRED_FILE_FOR_GRUB_CONFIG",
+                        title="Unable to find required file for GRUB config",
+                        description="Unable to find the original file required for GRUB configuration: %s" % src_path,
+                    )
+                    return
```

Comment:
We should break here? If more than one file does not exist, we would try to overwrite the set_result? Or it would end up in a failure because we already set an result?

Reply:
Good point. It probably would end in a failure if we set a result. Maybe I can use the flag from before and if it's true after the loop finishes then I can set the result

Reply:
@r0x0d do you think this block is fine as is since it will already end in a failure because we set a result?